### PR TITLE
Add Whisper benchmark script

### PR DESCRIPTION
**Pull request by kunal-vaishnavi**
### Description
This PR adds a benchmark script for Whisper. It is a follow-up to [this PR](https://github.com/microsoft/onnxruntime/pull/17020) that adds the LLaMA scripts.

### Motivation and Context
This PR enables benchmarking Whisper across various configurations.

### CODE REVIEW

**Comment by kunal-vaishnavi** [comment]
Splitting changes for LLaMA and Whisper into separate PRs as discussed offline

File: onnxruntime/python/tools/transformers/models/whisper/README.md [hunk 1]
```diff
@@ -55,3 +55,167 @@ $ python3 convert_to_onnx.py -m openai/whisper-tiny --output whispertiny --use_e
 # From wheel:
 $ python3 -m onnxruntime.transformers.models.whisper.convert_to_onnx -m openai/whisper-tiny --output whispertiny --use_external_data_format --precision int8 --quantize_embedding_layer
 ```
+
+## Benchmark Whisper
```
Comment (zhangyaobit) [inline onnxruntime/python/tools/transformers/models/whisper/README.md:59]: Could you also add A100 performance numbers for various versions of the model (tiny, medium, large)?

File: onnxruntime/python/tools/transformers/models/whisper/README.md [hunk 2]
```diff
@@ -55,3 +55,167 @@ $ python3 convert_to_onnx.py -m openai/whisper-tiny --output whispertiny --use_e
...
+### E2E variants
+
+1. PyTorch (without `torch.compile`), FP32, Hugging Face `pipeline` API
+```
+python3 benchmark.py \
+    --benchmark-type "HF + PT" \
+    --hf-api pipeline \
+    --audio-path 1272-141231-0002.mp3 \
+    --precision fp32 \
+    --model-size tiny \
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/models/whisper/README.md:74]: please use large model in example

File: onnxruntime/python/tools/transformers/models/whisper/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,696 @@
...
+    else:
+        run_ort_only_inference(args, inputs, model)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    # Args for benchmark type
+    parser.add_argument(
+        "-bt", "--benchmark-type", type=str, required=True, choices=["HF + PT", "HF + PT2", "HF + ORT", "ORT"]
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/models/whisper/benchmark.py:524]: Use one word like hf_pt, hf_pt2, hf_ort, ort. May add some description about what the meaning of each is.

File: onnxruntime/python/tools/transformers/models/whisper/README.md (hunk 1 above)
Comment (kunal-vaishnavi) [reply onnxruntime/python/tools/transformers/models/whisper/README.md:78]: Added

**Review by tianleiwu — APPROVED** [review]

**Comment by cloudhan** [comment]
#16895 is needed for ROCm EP to work as workaround. Otherwise, alot of patching is need for optimum.

**Pull request merged**